workspace/find-packages/src/index.ts

```diff
@@ -55,15 +61,64 @@ export async function findWorkspacePackagesNoCheck (workspaceRoot: string, opts?
   return pkgs
 }
 
-async function requirePackagesManifest (dir: string): Promise<{ packages?: string[] } | null> {
+async function requirePackagesManifest (dir: string): Promise<WorkspaceManifest | undefined> {
```

Comment:
I think it might be useful now to move this to a new package at `workspace/read-manifest` and name it `@pnpm/workspace.read-manifest`.

Reply:
Done :)